src/sentry/api/event_search.py

```diff
@@ -776,7 +776,8 @@ def to_list(value):
                     projects = get_projects(params)
                 condition = ["project_id", "=", projects.get(term.value.value)]
                 kwargs["conditions"].append(condition)
-            elif name == "issue.id":
+            elif name == "issue.id" and term.value.value != "":
```

Comment:
Not trying to make your life harder, but I am curious as to why we should special case issue.id here. I was operating under the assumption that this would be relevant to any/most fields... You can tell me to stay in my lane and faff off, but this is where my brain goes.

Reply:
@logicalbomb `Issue.id `is special cause we want to throw them in the `group_ids` bucket since having those can significantly speed up snuba queries.

But in the case of a having (where there isn't actually an id) we don't want to treat them special anymore, and at this point in the flow the query condition has become a `SearchFilter` which only has key, operator, value.

For something like a `has` this becomes something like `issue.id != ""` (ie. issue.id isn't nothing, so we "have" an issue.id) Which means when we hit this block we're trying to get that nice group id speed up, but on an empty string, which borks things.